common/lib/xmodule/xmodule/course_module.py

```diff
@@ -910,6 +918,48 @@ def __init__(self, *args, **kwargs):
             self.discussion_topics = {_('General'): {'id': self.location.html_id()}}
 
         try:
+            enrollment_track_scheme = UserPartition.get_scheme("enrollment_track")
+            enrollment_track_partitions = self.get_user_partitions_for_scheme(enrollment_track_scheme)
+            # Note that when a course_module is initially created, the ID is the draft version.
+            # We just want to store the course_id without published or draft.
+            course_id = unicode(self.id.for_branch(None))
+            create_partition = False
+            new_id = 0
+
+            if len(enrollment_track_partitions) == 0:
+                used_ids = set(p.id for p in self.user_partitions)
+                while new_id in used_ids:
+                    new_id += 1
+                create_partition = True
+            elif enrollment_track_partitions[0].parameters["course_id"] != course_id:
+                # With older courses, the first time through this code we may end up storing
+                # the "old mongo" version of the course key. If so, update it to the new version.
+                self.user_partitions.remove(enrollment_track_partitions[0])
+                new_id = enrollment_track_partitions[0].id
+                create_partition = True
+
+            if create_partition:
+                partition = enrollment_track_scheme.create_user_partition(
+                    id=new_id,
+                    name=_(u"Enrollment Track Partition"),
+                    description=_(u"Partition for segmenting users by enrollment track"),
+                    parameters={"course_id": course_id}
+                )
+                self.user_partitions.append(partition)
+
+            if len(enrollment_track_partitions) > 1:
+                log.warning(
+                    "Multiple EnrollmentTrackUserPartitions found for course {course_id}".format(
+                        course_id=unicode(self.id)
+                    )
+                )
+
+        except UserPartitionError:
```

Comment:
Sure, will make it just around UserPartition.get_scheme. This will probably also change soon as we are going to make the code robust to missing schemes (as part of the ICRV removal).